plugins/modules/proxmox.py

```diff
@@ -501,6 +615,130 @@ def update_config(self, vmid, node, disk, cpus, memory, swap, **kwargs):
+
+            return disk_kwargs
+
+        def convert_mounts(mount_dict):
+            return_list = []
+            for mount_key, mount_value in mount_dict.items():
+                mount_config = parse_disk_string(mount_value)
+                return_list.append(dict(id=mount_key, **mount_config))
+
+            return return_list
+
+        def build_volume(
+            key,
+            storage=None,
+            volume=None,
+            host_path=None,
+            size=None,
+            mountpoint=None,
+            options=None,
+            **kwargs
+        ):
+            if size is not None and isinstance(size, str):
+                size = size.strip("G")
+            # 1. Handle volume checks/creation
+            # 1.1 Check if defined volume exists
+            if volume is not None:
+                storage_content = self.get_storage_content(node, storage, vmid=vmid)
+                vol_ids = [vol["volid"] for vol in storage_content]
+                volid = "{storage}:{volume}".format(storage=storage, volume=volume)
+                if volid not in vol_ids:
+                    self.module.fail_json(
+                        changed=False,
+                        msg="Storage {storage} does not contain volume {volume}".format(
+                            storage=storage,
+                            volume=volume,
+                        ),
+                    )
+                vol_string = "{storage}:{volume},size={size}G".format(
+                    storage=storage, volume=volume, size=size
+                )
+            # 1.2 If volume not defined (but storage is), check if it exists
+            elif storage is not None:
+                api_node = self.proxmox_api.nodes(
+                    node
+                )  # The node must exist, but not the LXC
+                try:
+                    vol = api_node.lxc(vmid).get("config").get(key)
+                    volume = parse_disk_string(vol).get("volume")
+                    vol_string = "{storage}:{volume},size={size}G".format(
+                        storage=storage, volume=volume, size=size
+                    )
+
+                # If not, we have proxmox create one using the special syntax
+                except Exception:
+                    vol_string = "{storage}:{size}".format(storage=storage, size=size)
+
+            # 1.3 If we have a host_path, we don't have storage, a volume, or a size
+            if host_path is not None:
+                vol_string = host_path
+
+            # 2. Build valid string
+            if mountpoint is not None:
+                vol_string += ",mp={mountpoint}".format(mountpoint=mountpoint)
+            if options is not None:
+                vol_string += ","
+                vol_string += ",".join(map("=".join, options.items()))
+            if kwargs:
+                vol_string += ","
+                vol_string += ",".join(map("=".join, kwargs.items()))
```

Comment:
Another way to express that (feel free to ignore if you don't like it) would be:
```suggestion
            vol_string = ",".join(
                [] if host_path is None else [host_path] +
                [] if mountpoint is None else ["mp={0}".format(mountpoint)] +
                [] if options is None else [map("=".join, options.items())] +
                [] if not kwargs else [map("=".join, kwargs.items())]
            )
```
It is more concise, but not necessarily better.